expression/builtin_time_vec.go

```diff
@@ -1732,11 +1732,81 @@ func (b *builtinToDaysSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinDateFormatSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDateFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+
+	dateBuf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dateBuf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, dateBuf); err != nil {
+		return err
+	}
+	times := dateBuf.Times()
+
+	formatBuf, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(formatBuf)
+	if err := b.args[1].VecEvalString(b.ctx, input, formatBuf); err != nil {
+		return err
+	}
+
+	var dateAsIntBuf *chunk.Column
+	result.ReserveString(n)
+
+	for i := range times {
+		t := times[i]
+		if dateBuf.IsNull(i) || formatBuf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+
+		formatMask := formatBuf.GetString(i)
+		// MySQL compatibility, #11203
+		// If format mask is 0 then return 0 without warnings
+		if formatMask == "0" {
+			result.AppendString("0")
+			continue
+		}
+
+		if t.InvalidZero() {
```

Comment:
This branch is used to fix https://github.com/pingcap/tidb/issues/11203, and in this branch, it will call `b.args[0].VecEvalInt`(or `b.args[0].EvalInt` in the row-based implementation), which leads the CI problem since calling `VecEvalInt` on a time column is incorrect.

Reply:
I will create another to fix this problem soon, please wait for my PR.

Reply:
Please wait this PR https://github.com/pingcap/tidb/pull/15263.